CONTRIBUTING.md

```diff
@@ -1,34 +1,51 @@
-license document, but changing it is not allowed.
+Make sure to read our [Contributor License Agreement](CLA.md).
 
+## Getting Started
+### New Contributors
+We're excited to help you make your first contribution! Whether you're interested in filing issues, developing features, fixing bugs, or improving documentation, we're here to support you through the process.
 
-Developer's Certificate of Origin 1.1
+Browse issues labeled [good first issue] or [help wanted] on GitHub for an easy introduction.
 
-By making a contribution to this project, I certify that:
+### Developers
+The main building blocks of KAI Scheduler are documented in the `docs/developer` folder. Here are the key components:
+- [Action Framework](docs/developer/action-framework.md): Core scheduler logic
+- [Plugin Framework](docs/developer/plugin-framework.md): Extensible plugin system
+- [Pod Grouper](docs/developer/pod-grouper.md): Group scheduling functionality
+- [Binder](docs/developer/binder.md): Binding logic
 
-(a) The contribution was created in whole or in part by me and I
-    have the right to submit it under the open source license
-    indicated in the file; or
+We recommend reading these documents to understand the architecture before making significant contributions.
 
-(b) The contribution is based upon previous work that, to the best
-    of my knowledge, is covered under an appropriate open source
-    license and I have the right under that license to submit that
-    work with modifications, whether created in whole or in part
-    by me, under the same open source license (unless I am
-    permitted to submit under a different license), as indicated
-    in the file; or
+## How to Contribute
+### Reporting Issues
+Open an issue with a clear description, steps to reproduce, and relevant environment details.
+Use our issue templates for guidance.
 
-(c) The contribution was provided directly to me by some other
-    person who certified (a), (b) or (c) and I have not modified
-    it.
+### Improving Documentation
+Help us keep the docs clear and useful by fixing typos, updating outdated information, or adding examples.
 
-(d) I understand and agree that this project and the contribution
-    are public and that a record of the contribution (including all
-    personal information I submit with it, including my sign-off) is
-    maintained indefinitely and may be redistributed consistent with
-    this project or the open source license(s) involved.
+### Code & Documentation
+- Fork & Clone: Start by forking the repository and cloning it locally.
+- Create a Branch: Use a descriptive name (e.g., feature/add-cool-feature or bugfix/fix-issue123).
+- Make Changes: Keep commits small and focused. For detailed build and test instructions, see [Building from Source](docs/developer/building-from-source.md).
+- Submit a PR: Open a pull request, reference any related issues, and follow our commit message guidelines.
```

Comment:
Do we have any commit message guidelines? If we don't have it for now, let's skip it @romanbaron